src/shared/components/Tabs/Tabs.tsx

```diff
@@ -1,29 +1,64 @@
-import React, { useState } from 'react'
-import { TabsGroup, Tab } from './Tabs.styles'
+import React, { useEffect, useRef, useState } from 'react'
+import useResizeObserver from 'use-resize-observer'
+import { TabsGroup, Tab, TabsWrapper, BackgroundGradient, TAB_WIDTH } from './Tabs.styles'
 
-type TabsProps = {
+export type TabsProps = {
   tabs: string[]
   initialIndex?: number
   onSelectTab: (idx: number) => void
 }
 const Tabs: React.FC<TabsProps> = ({ tabs, onSelectTab, initialIndex = -1 }) => {
   const [selected, setSelected] = useState(initialIndex)
+  const [center, setCenter] = useState(0)
+  const tabsRef = useRef<HTMLDivElement>(null)
+  useResizeObserver<HTMLDivElement>({
+    ref: tabsRef,
+    onResize: ({ width }) => {
+      if (!tabsRef.current || !width) {
+        return
+      }
+      const isScrollBarVisible = tabsRef.current.scrollWidth > width
+      if (isScrollBarVisible) {
+        setCenter(width / 3)
+      } else {
+        setCenter(0)
+      }
+    },
+  })
+
+  useEffect(() => {
+    const tabsGroup = tabsRef.current
+    if (!tabsGroup) {
+      return
+    }
+    const currentItemOffsetleft = selected > 0 && TAB_WIDTH * selected
+    if (currentItemOffsetleft) {
+      tabsGroup.scrollLeft = currentItemOffsetleft - center
+    }
+  }, [center, selected, tabs.length, tabsRef])
+
+  const isLeftGradientVisible = center > 0 && selected > 0
+  const isRightGradientVisible = center > 0 && selected !== tabs.length - 1
 
   return (
-    <TabsGroup>
-      {tabs.map((tab, idx) => (
-        <Tab
-          onClick={(e) => {
-            onSelectTab(idx)
-            setSelected(idx)
-          }}
-          key={`${tab}-${idx}`}
-          selected={selected === idx}
-        >
-          <span>{tab}</span>
-        </Tab>
-      ))}
-    </TabsGroup>
+    <TabsWrapper>
+      {isLeftGradientVisible && <BackgroundGradient direction="prev" />}
+      <TabsGroup ref={tabsRef}>
+        {tabs.map((tab, idx) => (
+          <Tab
+            onClick={() => {
+              onSelectTab(idx)
+              setSelected(idx)
+            }}
+            key={`${tab}-${idx}`}
+            selected={selected === idx}
+          >
+            <span>{tab}</span>
+          </Tab>
+        ))}
+      </TabsGroup>
+      {isRightGradientVisible && <BackgroundGradient direction="next" />}
```

Comment:
I see we only display a right shadow which isn't ideal. If we use a shadow as an indication of "some more content is there" then we will need it on both sides once you scroll. It also should disappear when you scroll all the way to the right